### PR TITLE
[MO] add support for scalar shapes into cli_parser.py

### PR DESCRIPTION
**Pull request by pavel-esir**
#### Details:
 - corrected regular expression so that scalar shape in `--input_shape=inp[]` or `--input_shape=input1[1,100,100,3],input2[]`
 - `--input` argument already has support of scalar input shapes

#### Tickets:
 - xxx-99609

### CODE REVIEW

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 1]
```diff
@@ -1490,15 +1490,15 @@ def split_inputs(input_str):
...
+    first_digit_reg = r'([0-9 ]*|-1|\?|{})'.format(range_reg)
     next_digits_reg = r'(,{})*'.format(first_digit_reg)
     tuple_reg = r'((\({}{}\))|(\[{}{}\]))'.format(first_digit_reg, next_digits_reg,
                                                   first_digit_reg, next_digits_reg)
 
     full_reg = r'^{}(\s*,\s*{})*$|^$'.format(tuple_reg, tuple_reg)
     if not re.match(full_reg, argv_input_shape):
         raise Error('Input shape "{}" cannot be parsed. ' + refer_to_faq_msg(57), argv_input_shape)
-    return re.findall(r'[(\[]([0-9,\.\? -]+)[)\]]', argv_input_shape)
+    return re.findall(r'[(\[]([0-9,\.\? -]*)[)\]]', argv_input_shape)
```
Comment (popovaan) [inline tools/mo/openvino/tools/mo/utils/cli_parser.py:1501]: Do we need convert_model() to support following syntax?
convert_model(input_shape=[])
convert_model(input=InputCutInfo(shape=[]))

File: tools/mo/openvino/tools/mo/utils/cli_parser.py [hunk 2]
```diff
@@ -1579,7 +1579,7 @@ def get_placeholder_shapes(argv_input: str, argv_input_shape: str, argv_batch=No
         # clean inputs from values for freezing
         inputs_without_value = list(map(lambda x: x.split('->')[0], inputs))
         placeholder_shapes = dict(zip_longest(inputs_without_value,
-                                              map(lambda x: PartialShape(x) if x else None, shapes)))
+                                              map(lambda x: PartialShape(x) if (x or x == '') else None, shapes)))
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/utils/cli_parser.py:1582]: why is this change needed? Please test this case as well.

Comment (pavel-esir) [reply tools/mo/openvino/tools/mo/utils/cli_parser.py:1582]: This line is needed exactly for the cases with inputs with scalars and all new test cases test this line as well. E.g. if there are inputs `[12,4,1],[]` they will be split into 2 values with `'12,4,1'` and `''` and for the later case we need not `None` but `PartialShape([])` or equivalent `PartialShape('')`

image

File: tools/mo/openvino/tools/mo/utils/cli_parser.py (hunk 1 above)
Comment (pavel-esir) [reply tools/mo/openvino/tools/mo/utils/cli_parser.py:1501]: `convert_model` is already supporting inputs with scalar shapes. Added an test case for that

**Review by rkazants — APPROVED** [review]
